### PR TITLE
Refactor iosxr_config to make use of network_connection plugin

### PR DESCRIPTION
**Pull request by rcarrillocruz**
##### ISSUE TYPE

 - Feature Pull Request

##### COMPONENT NAME

iosxr_config

##### ANSIBLE VERSION

```
ansible 2.3.0 (refactor_ios_config_network_connection 784bd58c4a) last updated 2017/01/23 10:44:35 (GMT +200)
```

##### SUMMARY

Refactor iosxr_config to make use of network_connection plugin

### CODE REVIEW

File: lib/ansible/modules/network/iosxr/iosxr_config.py [hunk 1]
```diff
@@ -189,10 +177,27 @@
...
+  type: str
+  sample: "2016-11-16 10:38:25.595612"
+delta:
+  description: The time elapsed to perform all operations
+  returned: always
+  type: str
+  sample: "0:00:10.469466"
 """
+from ansible.module_utils.local import LocalAnsibleModule
 from ansible.module_utils.basic import get_exception
```
Comment (privateip) [inline lib/ansible/modules/network/iosxr/iosxr_config.py]: not needed

File: lib/ansible/modules/network/iosxr/iosxr_config.py [hunk 2]
```diff
@@ -260,7 +261,8 @@ def run(module, result):
 
             result['updates'] = commands
 
-        load_config(module, commands, result)
+        load_config(module, commands, not module.check_mode)
+        result['changed'] = not module.check_mode
```
Comment (privateip) [inline lib/ansible/modules/network/iosxr/iosxr_config.py]: Shouldn't the `changed` flag be based on the return.  If `diff` is empy `changed=False` else `changed=True`?

Comment (privateip) [reply lib/ansible/modules/network/iosxr/iosxr_config.py]: Also, if result should return the diff key with the diff in it to support calling `ansible-playbook` with the `--diff` option

File: lib/ansible/modules/network/iosxr/iosxr_config.py [hunk 3]
```diff
@@ -260,7 +261,8 @@ def run(module, result):
 
             result['updates'] = commands
 
-        load_config(module, commands, result)
+        load_config(module, commands, not module.check_mode)
```
Comment (privateip) [inline lib/ansible/modules/network/iosxr/iosxr_config.py]: `load_config` needs to support the `comment` argument

**Review by privateip — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/network/iosxr/iosxr_config.py (hunk 1 above)
Comment (rcarrillocruz) [reply lib/ansible/modules/network/iosxr/iosxr_config.py]: done

File: lib/ansible/modules/network/iosxr/iosxr_config.py (hunk 3 above)
Comment (rcarrillocruz) [reply lib/ansible/modules/network/iosxr/iosxr_config.py]: yep, cargo-culted from module_utils/ios.py which didn't have 'comment', will push a patch to fix it as well.

File: lib/ansible/modules/network/iosxr/iosxr_config.py (hunk 2 above)
Comment (rcarrillocruz) [reply lib/ansible/modules/network/iosxr/iosxr_config.py]: done

**Comment by privateip** [comment]
ship_it

**Review by privateip — APPROVED** [review]

**Pull request merged**